libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/SharedApp.kt

```diff
@@ -25,6 +26,8 @@ object SharedApp {
     val store = Store()
     val state get() = store.state.value
 
+    private val ignoreTripSessionUpdates = AtomicBoolean(false)
```

Comment:
If Store is for public state, is there a better place to put internal state?

Reply:
Will you add a comment to this? It seems the underlying issue where we need to "stop/start" sessions is the reason this is needed. But instead of a debounce we have a boolean guard.

```
        // TripSessionState is triggered rapidly when enabling and disabling replay by
        // calling `startTripSession, `stopTripSession`, `startReplayTripSession`
```

Reply:
There is no better place to put this flag in.

Reply:
> Will you add a comment to this? It seems the underlying issue where we need to "stop/start" sessions is the reason this is needed. But instead of a debounce we have a boolean guard.

The comment added to `SharedApp.tripSessionTransaction` is enough. I won't be adding comment to **this**.

Reply:
> There is no better place to put this flag in.

yeah, I think that it should not be needed in the first place 🤔 .

For example, the reason a restart is needed is because [MapboxTripSession](https://github.com/mapbox/mapbox-navigation-android/blob/0104295d6d203c97718c208532010c77d864a9b4/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L239-L241) ignores sessions that are started. But it should be able to detect if the internal state is changing.

Reply:
This problem has more to do with the fact that when we rotate the screen, `TripSessionComponent` being a `UIComponent` detaches thereby invoking `tripSession.stop()`. Wouldn't it be better to move this to a state controller? 

Not blocking this PR since we are unable to test screen rotation at the moment. However this is something we should see if we can improve.

Reply:
`UIComponents` are tied to `NavigationView` lifecycle, while `StateControllers` are to `MapboxNavigation` lifecycle.
If we move this `StateController`, we'll have to update how the Android Audio module manages TripSession.

Reply:
Let's think about this more and do this refactoring as a follow up to this PR.